api/rtx/src/unsupported_malloc.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2015-2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "uvisor-lib/uvisor-lib.h"
+
+#if !(defined(UVISOR_PRESENT) && (UVISOR_PRESENT == 1))
+
+#include <string.h>
+
+/* Note: This file is not included in the uVisor release library. Instead, the
+ *       host OS needs to compile it separately if a platform does not support
+ *       uVisor (but uVisor API header files are still used). */
+
+void page_allocator_init(void * const heap_start, void * const heap_end, const uint32_t * const page_size);
+int page_allocator_malloc(UvisorPageTable * const table);
+int page_allocator_free(const UvisorPageTable * const table);
+
+extern uint32_t __end__[];      /* __heap_start */
+extern uint32_t __HeapLimit[];  /* __heap_end   */
+
+extern uint32_t __StackLimit[];   /* bottom of stack */
+
+RtxBoxIndex * __uvisor_ps;
+
+static void box_index_init(void *box_bss, uint32_t heap_size)
```

Comment:
This function seems like a copy of some code I reviewed earlier, for inside uVisor. Be sure it is in sync with that implementation, if any improvements were made in the uVisor implementation.